Task2/Kuba/NoughtsAndCrossesWPF/NoughtsAndCrossesWPF/ViewModel/ViewModel.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Windows.Input;
+
+namespace NoughtsAndCrossesWPF
+{
+    public class ViewModel //I'm not sure if I have put correct methods in this class. I feel like it's a bit messed but have no better idea how to organize this.
+    {
+        public string[] translatedOneDimensionalBoard = new string[GameBoard.boardSize * GameBoard.boardSize];
+        private ObservableCollection<string> _ObservableBoard;
+        public ObservableCollection<string> ObservableBoard { get { return _ObservableBoard; } }
+        static WPFBoardCommunication communicator = new WPFBoardCommunication();
+        GameBoard board = new GameBoard(); 
+        Game game = new Game(communicator);
+        Player player1 = new Player(GameBoardStatus.cross, "Crosses"); //TODO: implement player's name input box
+        Player player2 = new Player(GameBoardStatus.nought, "Noughts");
+        Player currentPlayer;
+
+        public ViewModel()
+        {
+            _ObservableBoard = new ObservableCollection<string>(translatedOneDimensionalBoard);
+            currentPlayer = player1;
+        }
+
+        private void translateBoardToOneDimensional(GameBoard board)
+        {
+            for (int i = 0; i < translatedOneDimensionalBoard.Length; i++)
+            {
+                Position coordinates = communicator.TranslateOneDimensionalPosition(i);
+                translatedOneDimensionalBoard[i] = TranslateBoardSymbol(board.gameBoard[coordinates.Vertical, coordinates.Horizontal]);
+            }
+        }
+
+        private string TranslateBoardSymbol(GameBoardStatus symbol)
+        {
+            if (symbol == GameBoardStatus.empty) return "";
+            if (symbol == GameBoardStatus.cross) return "X";
+            return "O";
+        }
+
+        private void UpdateObservableBoard()
+        {
+            for (int i = 0; i < 9; i++)
+            {
+                ObservableBoard[i] = translatedOneDimensionalBoard[i];
+            }
+        }
+
+        private ICommand _clickCommand;
+        private bool _canExecute = true;
+
+        public ICommand ClickCommand
```

Comment:
the idea of Command is that it is independent on how it was executed. Click suggest that it is executed by a click (when it could be also Hotkey, Joystick, Steering wheel, Voice Command, Web API or whatever). I would make it something like MarkFieldCommand or MakeAMoveCommand